### ISSUE TITLE
UnsupportedOperationException while invoking Statement.close with jdbc-4.2

### ISSUE THREAD

**Issue by lmsurpre** [OpenLiberty/open-liberty#5240]
I'm using liberty 18.0.0.2 with the `jdbc-4.2` feature and a manually-configured XADataSource from the IBM Data Server Driver for JDBC and SQLJ (`com.ibm.db2.jcc.DB2XADataSource` from `db2jcc4.jar`) and I'm seeing an issue very similar to issue #3700.  Specifically, I can instantiate and invoke a PreparedStatement, but when I try to close that statement I'm seeing an error like this:

```MyException...
Caused by: java.lang.UnsupportedOperationException: getLargeUpdateCount not implemented
  at java.sql.Statement.getLargeUpdateCount(Statement.java:1103)
  at com.ibm.ws.rsadapter.jdbc.v42.WSJdbc42PreparedStatement.getCompatibleUpdateCount(WSJdbc42PreparedStatement.java:54)
  at com.ibm.ws.rsadapter.impl.DatabaseHelper.getUpdateCount(DatabaseHelper.java:336)
  at com.ibm.ws.rsadapter.jdbc.WSJdbcPreparedStatement.closeWrapper(WSJdbcPreparedStatement.java:306)
  at com.ibm.ws.rsadapter.jdbc.WSJdbcObject.close(WSJdbcObject.java:152)
  at com.ibm.ws.rsadapter.jdbc.WSJdbcObject.close(WSJdbcObject.java:110)
  at MyClass.cleanup()
```

Whats odd to me is that I don't even see the db2 driver in that stack trace.  However, when I drop the liberty feature down to `jdbc-4.1`, I can see it using the proper driver and everything is working.

Either way, from #3700 it sounds like the liberty jdbc wrapper is supposed to be tolerant of drivers which havn't implemented getLargeUpdateCount ...so any idea why its throwing this exception during cleanup?

**Comment by aguibert** [comment]
hi @lmsurpre, this issue is fixed in 18.0.0.3 via PR #4198

**Comment by aguibert** [comment]
Closing this issue as fixed in 18.0.0.3.

For now, the workarounds are:
 A) downgrade to `jdbc-4.1` Liberty feature
 B) upgrade to a JDBC 4.2 compliant JDBC driver (which is not possible for DB2 as they do not support JDBC 4.2 yet)
